frontend/src/pages/PipelineDetailsV1.tsx

```diff
@@ -260,7 +261,9 @@ const PipelineDetailsV1: React.FC<PipelineDetailsV1Props> = ({
         )}
         {selectedTab === 1 && !!templateString && (
           <div className={css.containerCss} data-testid={'spec-yaml'}>
-            <PipelineSpecTabContent templateString={templateString || ''} />
+            <PipelineSpecTabContent
+              templateString={jsyaml.safeDump(jsyaml.safeLoad(templateString)) || ''}
```

Comment:
We should add some documentations here to explain why do we dump and load the templateString. People who read this change cannot find any reference in tests, either. 

In the documentation, explaining the followings is important:
- The input templateString can be a `json formatted` string
- SafeLoad can convert the string to object
- SafeDump will change the object to yaml format
- yaml format is acceptable by PipelineSpecTabContent